apps/application/flow/step_node/question_node/impl/base_question_node.py

```diff
@@ -138,7 +140,7 @@ def get_details(self, index: int, **kwargs):
             'name': self.node.properties.get('stepName'),
             "index": index,
             'run_time': self.context.get('run_time'),
-            'system': self.node_params.get('system'),
+            'system': self.context.get('system'),
             'history_message': [{'content': message.content, 'role': message.type} for message in
                                 (self.context.get('history_message') if self.context.get(
                                     'history_message') is not None else [])],
```

Comment:
There are a few issues in the code you've provided:

1. **Incorrect Use of `workflow_manage`**: The line `system = self.workflow_manage.generate_prompt(system)` seems to be intended to generate a new system string based on existing one, but there's no context about where this method exists. It should either belong to another class instance or be part of an imported module.

2. **Redundant Context Update**: You're updating the context with the same value twice for `system`: once at the beginning (`self.context['system']`) and again after generating it (`if stream:` block). This might introduce redundancy unless the workflow manages both contexts differently.

3. **List Comprehension Formatting**: In both instances where you create lists from dictionaries, consider using more compact syntax for readability, e.g., `[{'content': m.content, 'role': m.type} for m in messages]`.

Overall, make sure that `workflow_manage` has its own methods appropriately defined and importable within your current script so that these lines can work without errors. If they need to use external data structures or functions, ensure those components are available and properly structured according to their requirements.